File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/utils/SharePreferencesUtil.java
package com.lilanz.tooldemo.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class SharePreferencesUtil {

    private static final String SHARED_PREFERENCE_NAME = "shared";
    private static final String PARTERN_ID = "parternId";   // 合作者Id

    private static SharedPreferences instance = null;


    // 获取合作者id
    public static int getParternId(Context context) {
        return getInstance(context).getInt(PARTERN_ID, 0);
    }

    // 保存合作者id
    public static void setParternId(Context context, int id) {
        setInteger(context, PARTERN_ID, id);
    }

    private static final String SELECT_BLE_ADDRESS = "selectedBleAddress";   //

    // 获取选择的蓝牙地址
    public static String getBleAddress(Context context) {
        return getInstance(context).getString(SELECT_BLE_ADDRESS, "");
    }

    public static void saveBleAddress(Context context, String address) {
        setString(context, SELECT_BLE_ADDRESS, address);
    }

    private static final String LAST_SAVE_DAY = "lastSaveDay";   //

    // 获取
    public static String getLastSaveDay(Context context) {
        return getInstance(context).getString(LAST_SAVE_DAY, "");
    }

    public static void saveLastSaveDay(Context context, String dayStr) {
        setString(context, LAST_SAVE_DAY, dayStr);
    }

    //============================      以下为固定模式        =============================

    // 保存int型数据
    private static void setInteger(Context context, String key, int value) {
        SharedPreferences.Editor editor = getInstance(context).edit();
        editor.putInt(key, value);
        editor.commit();
    }

    // 保存float型数据
    private static void setFloat(Context context, String key, float value) {
        SharedPreferences.Editor editor = getInstance(context).edit();
        editor.putFloat(key, value);
        editor.commit();
    }

    // 保存boolean型数据
    private static void setBoolean(Context context, String key, boolean b) {
        SharedPreferences.Editor editor = getInstance(context).edit();
        editor.putBoolean(key, b);
        editor.commit();
    }

    // 保存String型数据
    private static void setString(Context context, String key, String value) {
        SharedPreferences.Editor editor = getInstance(context).edit();
        editor.putString(key, value);
        editor.commit();
    }

    // 保存Long型数据
    private static void setLong(Context context, String key, long value) {
        SharedPreferences.Editor editor = getInstance(context).edit();
        editor.putLong(key, value);
        editor.commit();
    }

    private static SharedPreferences getInstance(Context context) {
        if (instance == null) {
            instance = context.getSharedPreferences(SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE);
        }
        return instance;
    }

}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/utils/SharePreferencesUtil.kt
package com.lilanz.kotlintool.utils

import android.content.Context
import android.content.SharedPreferences

class SharePreferencesUtil {
    companion object {
        private val SHARED_NAME: String = "shared"  // shared的文件名称
        private val PARTEN_ID: String = "partenId"  // 合作者id数据

        private lateinit var instance: SharedPreferences


        // 获取：合作者id
        fun getPartenId(context: Context): Int {
            return getInstance(context).getInt(PARTEN_ID, 0)
        }

        // 保存：合作者id
        fun setPartenId(context: Context, id: Int) {
            setInt(context, PARTEN_ID, id)
        }

        //=====================         一下为固定方法         ==========================

        /**
         * 保存boolean型数据
         */
        private fun setBoolean(context: Context, key: String, value: Boolean) {
            var editor = getInstance(context).edit()
            editor.putBoolean(key, value)
            editor.commit()
        }

        /**
         * 保存int型数据
         */
        private fun setInt(context: Context, key: String, value: Int) {
            var editor = getInstance(context).edit()
            editor.putInt(key, value)
            editor.commit()
        }

        /**
         * 保存String型数据
         */
        private fun setString(context: Context, key: String, value: String) {
            val editor = getInstance(context).edit()
            editor.putString(key, value)
            editor.commit()
        }

        /**
         * 获取SharedPreferences单例对象
         */
        private fun getInstance(context: Context): SharedPreferences {
            if (instance == null) {
                instance = context.getSharedPreferences(SHARED_NAME, Context.MODE_PRIVATE)
            }
            return instance
        }
    }
}